### PR TITLE
module: MALT/RUN

### LINKED ISSUE #636: new module: MALT
I think it would be good to have a module for [MALT(https://bioconda.github.io/recipes/malt/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
Adds the MALT run module

REQUIRES #642 and #645  to be merged first!

## PR checklist

Closes #636 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/malt/run/meta.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      documentation: https://software-ab.informatik.uni-tuebingen.de/download/malt/manual.pdf
+      tool_dev_url: None
+      doi: "10.1038/s41559-017-0446-6"
+      licence: ['GPL v3']
+
+input:
+  - fastqs:
+      type: file
+      description: Input FASTQ files
+      pattern: "*.{fastq.gz,fq.gz"
```
Comment (grst) [inline modules/malt/run/meta.yml:25]:
```suggestion
      pattern: "*.{fastq.gz,fq.gz}"
```

File: tests/modules/malt/run/main.nf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+include { UNZIP } from '../../../../modules/unzip/main.nf' addParams( options: [:] )
+include { MALT_BUILD } from  '../../../../modules/malt/build/main.nf' addParams( options: [:] )
+include { MALT_RUN } from '../../../../modules/malt/run/main.nf' addParams( options: [:] )
+
+workflow test_malt_run {
+
+    fastas = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    gff = file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true)
+    seq_type = "DNA"
+    map_db = file("https://software-ab.informatik.uni-tuebingen.de/download/megan6/megan-nucl-Jan2021.db.zip", checkIfExists: true)
```
Comment (grst) [inline tests/modules/malt/run/main.nf:14]: I don't know this tool... but would it be possible to create a small database based on toy data? 

If not, or if complicated, the CI seems still be able to cope with this in a reasonable time, so maybe not so important.

Comment (jfy133) [reply tests/modules/malt/run/main.nf:14]: In this case not possible unfortunately, it's a custom SQL version of NCBI taxdump (AFAIK)...

The module won't be updated that regularly though so it shouldn't block stuff too often.

Comment (grst) [reply tests/modules/malt/run/main.nf:14]: Ok, let's go then!

**Pull request merged**